picoshare/guest_link.go

```diff
@@ -14,7 +14,8 @@ type (
 		ID             GuestLinkID
 		Label          GuestLinkLabel
 		Created        time.Time
-		Expires        ExpirationTime
+		UrlExpires     ExpirationTime
+		FileExpires    string
```

Comment:
I had this as ExpirationTime in my first commit. The problem I saw there was the database entry was a timestamp and the file expiry for a guest link decreased with time.  (So if the guest link had a file expiration of 30 days, after 5 days all uploaded files would only be valid for 25 days, if the url expiry was greater than the file expiration this might lead to negative values)
But there is probably some better way of ensuring a fixed value in the database, I just couldn't think of it.

Reply:
Ah, I see what you're saying. 

Yeah, you're right. It can't be an `ExpirationTime` because it's not a specific point in time but rather a delta from whenever a file is uploaded to that guest link.

Can we define a new type like this?

```go
type (
	ExpirationTime time.Time
        FileLifetime   time.Duration
)

// Treat a distant expiration time as sort of a sentinel value signifying a "never expire" option.
var NeverExpire = ExpirationTime(time.Date(2999, time.December, 31, 0, 0, 0, 0, time.UTC))

// Treat the max possible duration as infinite lifetime.
var InfiniteLifetime = FileLifeTime(time.Duration(math.MaxInt64))
```

And then add a function `parse.FileLifetime` (similar to [`parse.Expiration`](https://github.com/mtlynch/picoshare/blob/bf6fe300e23e3b64f5d70f5fba55fb354892cb02/handlers/parse/expiration.go#L16)) that parses it from a string (using `time.ParseDuration`) and makes sure it's a valid value (e.g., not less than 1 hour)?

Also, in that case, when we're deserializing from the database, I think we should just treat `NULL` as `InfiniteLifetime` rather than having to recognize the special value `NEVER`.